icd/icd_context.h

```diff
@@ -4,30 +4,30 @@
 #include <glib.h>
 
 struct icd_context {
-  gboolean daemon;
-  guint shutting_down;
-  GMainLoop *main_loop;
+  gboolean daemon; //run as daemon if TRUE, run in foreground if FALSE
```

Comment:
it should be (beware that comments start with /** , not with /*):

/** run as daemon if TRUE, run in foreground if FALSE */
gboolean daemon;

/** icd shutdown timeout id */
guint shutting_down;

the same goes for the others. Also, add a new line after each member variable.